src/components/AttachmentModal.js

```diff
@@ -78,164 +78,153 @@ const defaultProps = {
     onModalHide: () => {},
 };
 
-class AttachmentModal extends PureComponent {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            isModalOpen: false,
-            shouldLoadAttachment: false,
-            isAttachmentInvalid: false,
-            isAuthTokenRequired: props.isAuthTokenRequired,
-            attachmentInvalidReasonTitle: null,
-            attachmentInvalidReason: null,
-            source: props.source,
-            modalType: CONST.MODAL.MODAL_TYPE.CENTERED_UNSWIPEABLE,
-            isConfirmButtonDisabled: false,
-            confirmButtonFadeAnimation: new Animated.Value(1),
-            file: props.originalFileName
```

Comment:
Coming from https://github.com/Expensify/App/issues/28036:
We could have updated `file` state value whenever `originalFileName` prop changes in useEffect.

Reply:
The root cause is [here](https://github.com/Expensify/App/issues/28036#issuecomment-1765250730)